wallet/src/Pos/Wallet/Web/Pending/Submission.hs

```diff
@@ -166,9 +174,63 @@ submitAndSavePtx submitTx PtxSubmissionHandlers{..} ptx@PendingTx{..} = do
         -- while transaction creation failed, due to protocol error or bug,
         -- then we better not remove this pending transaction
         void $ usingPtxCoords removeOnlyCreatingPtx ptx
+
     badInitPtxCondition = InternalError "Expected PtxCreating as initial pending condition"
 
     reportSubmitted ack =
         logDebug $
         sformat ("submitAndSavePtx: transaction submitted with confirmation?: "
                 %build) ack
+
+
+-- As simple and as clear as I could.
+submitAndSavePtxStates
+    :: TxSubmissionMode ctx m
+    => PendingTx
+    -> Timestamp
+    -> ((TxId, TxAux) -> m TxSubmissionResult)
+    -> m TxSubmissionResult
+submitAndSavePtxStates ptx@PendingTx{..} now mSaveTx =
+
+    case _ptxCond of
+
+      PtxApplying poolInfo -> do
+          let Just creationTime = poolInfo ^. thTimestamp
```

Comment:
You can't write in this way, it fails in `Nothing` case

Reply:
Hint: `case` supports guards as well:
```
case _ptxCond of
   PtxApplying poolInfo | 
   Just creationTime = poolInfo ^. thTimestamp,
   diffTimestamp new creationTime  > hour
```
I'm sorry in advance if referencing obvious thing

Reply:
Right, but will we ever have a case where we have `PtxApplying` and no `Timestamp`?
I thought about that and it seems that if a transaction is applying, it should have a timestamp, right? Otherwise, we have a transaction that doesn't have `creationTime` which is weird, to say the least, and it should fail?

I can change it, though.

Reply:
That's correct, any transaction tracked by re-submitter should have timestamp.
But this code is unsafe - if assumption ^ would stop holding one day or there is a bug, submission will totally break with uncertain error.

Reply:
Ok, will throw if `Nothing`. 
If that assumption stops holding one day, we really are in trouble either way.